axum/src/extract/request_parts.rs

```diff
@@ -56,7 +56,13 @@ where
     async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
         let uri = Extension::<Self>::from_request(req)
             .await
-            .unwrap_or_else(|_| Extension(OriginalUri(req.uri().clone())))
+            .unwrap_or_else(|_| {
+                #[cfg(debug_assertions)]
+                panic!("`OriginalUri` extension should always be present, but it wasn't");
```

Comment:
The other panic makes sense to me but I'm unsure about this one. There is the possibility of `ExtensionsAlreadyExtracted`, which would be a programmer error rather than a bug in axum. Should we maybe change the `Rejection` type in 0.5?

Also it's easy to call this `FromRequest` impl on a `RequestParts` that wasn't created through the axum `Router`. As a somewhat crazy example, somebody could have their own router that uses `FromRequest`.

Reply:
Thats a good point! I'll remove this one